android/src/main/java/com/tailscale/ipn/ui/view/MainView.kt

```diff
@@ -252,16 +257,35 @@ fun ConnectView(
     state: Ipn.State,
     user: IpnLocal.LoginProfile?,
     connectAction: () -> Unit,
-    loginAction: () -> Unit
+    loginAction: () -> Unit,
+    isAdmin: Boolean,
 ) {
+  val handler = LocalUriHandler.current
+
   Row(horizontalArrangement = Arrangement.Center, modifier = Modifier.fillMaxWidth()) {
     Column(horizontalAlignment = Alignment.CenterHorizontally, modifier = Modifier.fillMaxWidth()) {
       Column(
           modifier = Modifier.padding(8.dp).fillMaxWidth(0.7f).fillMaxHeight(),
           verticalArrangement = Arrangement.spacedBy(8.dp, alignment = Alignment.CenterVertically),
           horizontalAlignment = Alignment.CenterHorizontally,
       ) {
-        if (state != Ipn.State.NeedsLogin && user != null && !user.isEmpty()) {
+        if (state == Ipn.State.NeedsMachineAuth) {
+          Text(
+              text = stringResource(id = R.string.machine_auth_required),
+              style = MaterialTheme.typography.titleMedium,
+              textAlign = TextAlign.Center)
+          Text(
+              text = stringResource(id = R.string.machine_auth_explainer),
+              style = MaterialTheme.typography.bodyMedium,
+              textAlign = TextAlign.Center)
+           if(isAdmin) {
+            PrimaryActionButton(onClick = {  handler.openUri(Links.ADMIN_URL) }) {
```

Comment:
It would be nice to link directly to the machine in question. It would also be nice to use our login flow so that once the machine is approved, we automatically come back to the app.

Reply:
Nice suggestion btw...  @agottardo One more thing to add to the growing "Things Android does better that we need to fix on Apple" backlog...